yet-another-docker-plugin/src/main/java/com/github/kostyasha/yad/DockerCloud.java

```diff
@@ -221,12 +221,13 @@ private DockerSlave provisionWithWait(DockerSlaveTemplate template, Provisioning
         final DockerContainerLifecycle dockerContainerLifecycle = template.getDockerContainerLifecycle();
         final String imageId = dockerContainerLifecycle.getImage();
         final DockerComputerLauncher computerLauncher = template.getLauncher();
+        final String osType = this.getClient().infoCmd().exec().getOsType();
 
         //pull image
         dockerContainerLifecycle.getPullImage().exec(getClient(), imageId);
 
         LOG.info("Trying to run container for {}", imageId);
-        final String containerId = runContainer(template);
+        final String containerId = runContainer(template, osType);
```

Comment:
after image pull you can get info about image and store it in template. AFAIK image os type is specific to image, not to daemon.